qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/IgnoredMetaFieldAggregationIT.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.upgrades;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.IndexVersions;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class IgnoredMetaFieldAggregationIT extends ParameterizedRollingUpgradeTestCase {
+
+    private static String aggFirstIndex;
+    private static boolean aggFirstCreated;
+    private static String aggSecondIndex;
+    private static String runtimeFieldFirstIndex;
+    private static boolean runtimeFieldFirstCreated;
+    private static String runtimeFieldSecondIndex;
+
+    public IgnoredMetaFieldAggregationIT(@Name("upgradedNodes") int upgradedNodes) {
+        super(upgradedNodes);
+    }
+
+    @Before
+    public void setup() {
+        aggFirstCreated = false;
+        runtimeFieldFirstCreated = false;
+        aggFirstIndex = "aaa_" + randomAlphaOfLength(8).toLowerCase(Locale.ROOT);
+        aggSecondIndex = "bbb_" + randomAlphaOfLength(9).toLowerCase(Locale.ROOT);
+        runtimeFieldFirstIndex = "ccc_" + randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        runtimeFieldSecondIndex = "ddd_" + randomAlphaOfLength(11).toLowerCase(Locale.ROOT);
+    }
+
+    public void testAggregation() throws IOException {
+        if (isOldCluster()) {
+            assertRestStatus(client().performRequest(createNewIndex(aggFirstIndex)), RestStatus.OK);
+            assertRestStatus(client().performRequest(indexDocument(aggFirstIndex, "foofoo", "1024.12.321.777")), RestStatus.CREATED);
+            if (getOldClusterIndexVersion().before(IndexVersions.DOC_VALUES_FOR_IGNORED_META_FIELD)) {
+                assertAggregateIgnoredMetadataFieldException(
+                    aggFirstIndex,
+                    "Fielddata is not supported on field [_ignored] of type [_ignored]"
```

Comment:
The type is off in the message?

Reply:
The content type is kind of unexpected, as we set it to `_ignored`. But not a huge deal, I was not aware, lets' leave it as-is.